File: repository/src/ch/uzh/ifi/feedback/repository/model/Feedback.java
package ch.uzh.ifi.feedback.repository.model;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

import ch.uzh.ifi.feedback.library.rest.annotations.DbAttribute;
import ch.uzh.ifi.feedback.library.rest.annotations.DbIgnore;
import ch.uzh.ifi.feedback.library.rest.annotations.Serialize;
import ch.uzh.ifi.feedback.library.rest.validation.Id;
import ch.uzh.ifi.feedback.library.rest.validation.NotNull;
import ch.uzh.ifi.feedback.repository.serialization.FeedbackSerializationService;

@Serialize(FeedbackSerializationService.class)
public class Feedback {

	@NotNull
	private String title;
	@Id
	private Integer id;

	@NotNull
	@DbAttribute("application_id")
	private Integer applicationId;

	@DbAttribute("user_identification")
	@NotNull
	private String user;

	@DbAttribute("created_at")
	private Timestamp createdAt;

	@DbAttribute("updated_at")
	private Timestamp updatedAt;

	@DbAttribute("config_version")
	@NotNull
	private Double configVersion;
	private String language;

	@DbAttribute("context_information_id")
	private ContextInformation contextInformation;

	@DbIgnore
	private List<TextFeedback> textFeedbacks;
	@DbIgnore
	private List<RatingFeedback> ratings;
	@DbIgnore
	private List<ScreenshotFeedback> screenshots;
	@DbIgnore
	private List<FeedbackComment> feedbackComments;
	@DbIgnore
	private List<AttachmentFeedback> attachmentFeedbacks;
	@DbIgnore
	private List<AudioFeedback> audioFeedbacks;
	@DbIgnore
	private List<CategoryFeedback> categoryFeedbacks;
}

File: reporting_frontend/src/client/app/shared/models/feedbacks/feedback.ts
import {TextFeedback} from './text_feedback';
import {RatingFeedback} from './rating_feedback';
import {ScreenshotFeedback} from './screenshot_feedback';
import {CategoryFeedback} from './category_feedback';
import {PushConfiguration} from '../configurations/push_configuration';
import {mechanismTypes} from '../mechanisms/mechanism_types';
import {ConfigurationInterface} from '../configurations/configuration_interface';
import {FeedbackStatus} from './feedback_status';
import {AttachmentFeedback} from './attachment_feedback';
import {AudioFeedback} from './audio_feedback';


const validationMessages = {
  textMechanism: {
    noText: 'Please input a text'
  }
};


export class Feedback {
  id:number;
  title:string;
  userIdentification:string;
  language:string;
  applicationId:number;
  configurationId:number;
  configuration:ConfigurationInterface;
  ratingFeedbacks:RatingFeedback[];
  textFeedbacks:TextFeedback[];
  screenshotFeedbacks:ScreenshotFeedback[];
  categoryFeedbacks:CategoryFeedback[];
  audioFeedbacks:AudioFeedback[];
  attachmentFeedbacks:AttachmentFeedback[];
  createdAt:Date;
  personalFeedbackStatus:FeedbackStatus;
  read:boolean;

  constructor(title:string, userIdentification:string, language:string, applicationId:number, configurationId:number,
              ratingFeedbacks?:RatingFeedback[], textFeedbacks?:TextFeedback[],
              screenshotFeedbacks?:ScreenshotFeedback[], categoryFeedbacks?:CategoryFeedback[],
              audioFeedbacks?:AudioFeedback[], attachmentFeedbacks?:AttachmentFeedback[]) {
    this.title = title;
    this.userIdentification = userIdentification;
    this.language = language;
    this.applicationId = applicationId;
    this.configurationId = configurationId;
    this.ratingFeedbacks = ratingFeedbacks;
    this.textFeedbacks = textFeedbacks;
    this.screenshotFeedbacks = screenshotFeedbacks;
    this.categoryFeedbacks = categoryFeedbacks;
    this.audioFeedbacks = audioFeedbacks;
    this.attachmentFeedbacks = attachmentFeedbacks;
  }
}

